src/x11.cc

```diff
@@ -780,14 +781,14 @@ void x11_init_window(lua::state &l, bool own) {
 
   XFlush(display);
 
+  int root_mask = SubstructureNotifyMask;
+  XSelectInput(display, window.root, root_mask);
+
   int64_t input_mask = ExposureMask | PropertyChangeMask;
 #ifdef OWN_WINDOW
-  if (own_window.get(l)) {
-    input_mask |= StructureNotifyMask;
 #if !defined(BUILD_XINPUT)
-    input_mask |= ButtonPressMask | ButtonReleaseMask;
+  if (own_window.get(l)) { input_mask |= ButtonPressMask | ButtonReleaseMask; }
 #endif
-  }
```

Comment:
Did this too soon. Anyway, this is supposed to add window change watching in one of future commits to update window bounds caching and fix #1852